### PR TITLE
Add basic type definitions to SDK - Closes #4496

### LINKED ISSUE #4496: Add basic type definitions for lisk-sdk
### Description
Add basic type definitions for lisk-sdk.
### Motivation

Right now when using `Configurator` or `Application` classes exported from `lisk-sdk` package, IDE is unable to infer which methods those classes expose. Therefore, adding basic type definitions (TS) for those classes would make Lisk SDK easier to use.
### Acceptance Criteria
- Application class exposes public interfaces
- Configurator class exposes public interfaces

### PR DESCRIPTION
**Pull request by limiaspasdaniel**
### What was the problem?
Refer to #4496 
### How did I solve it?
Refer to #4496 
### Review checklist
- [ ] The PR resolves #4496 
- [ ] All new code is covered with unit tests
- [ ] Relevant integration / functional tests are added
- [ ] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [ ] Documentation has been added/updated

### CODE REVIEW

File: sdk/src/index.d.ts [hunk 1]
```diff
@@ -29,7 +46,7 @@ export {
 	Application,
 	version,
 	systemDirs,
-	configurator,
+	Configurator,
```
Comment (shuse2) [inline sdk/src/index.d.ts]: this configurator is not a class. it was exporting default configs

File: sdk/src/index.d.ts [hunk 2]
```diff
@@ -17,10 +17,27 @@ import * as BigNum from '@liskhq/bignum';
 import * as cryptography from '@liskhq/lisk-cryptography';
 import * as transactions from '@liskhq/lisk-transactions';
 
-declare const Application: any;
+declare class Application {
+	constructor(genesisBlock: object, config: object);
+	run(): Promise<void>;
+	registerTransaction(transactionClass: object): void;
```
Comment (shuse2) [inline sdk/src/index.d.ts]: is class `object`?

Comment (limiaspasdaniel) [reply sdk/src/index.d.ts]: True, either should be a function or we can set it to `any`?

File: sdk/src/index.d.ts (hunk 1 above)
Comment (limiaspasdaniel) [reply sdk/src/index.d.ts]: So it is an instance of `Configurator`?

File: sdk/src/index.d.ts (hunk 2 above)
Comment (shuse2) [reply sdk/src/index.d.ts]: I think for now we can set it to `any`, but probably it's good to find solution when we go to full typescript

File: sdk/src/index.d.ts (hunk 1 above)
Comment (shuse2) [reply sdk/src/index.d.ts]: yes!

File: sdk/src/index.d.ts (hunk 2 above)
Comment (shuse2) [reply sdk/src/index.d.ts]: Also, second parameter is optional matcher

File: sdk/src/index.d.ts [hunk 3]
```diff
@@ -17,11 +17,29 @@ import * as BigNum from '@liskhq/bignum';
 import * as cryptography from '@liskhq/lisk-cryptography';
 import * as transactions from '@liskhq/lisk-transactions';
 
-declare const Application: any;
+declare class Application {
+	constructor(genesisBlock: object, config: object);
+	run(): Promise<void>;
```
Comment (shuse2) [inline sdk/src/index.d.ts]: also missing access to `logger` and `config` for the `core`

File: sdk/src/index.d.ts [hunk 4]
```diff
@@ -17,11 +17,44 @@ import * as BigNum from '@liskhq/bignum';
...
+type ModuleClass = AnyClass;
+
+declare class Application {
+	readonly logger: any;
+	readonly config: any;
+
+	constructor(genesisBlock: any, config?: any);
+	run(): Promise<void>;
+	registerTransaction(
+		transactionClass: TransactionClass,
```
Comment (shuse2) [inline sdk/src/index.d.ts]: I got `Argument of type 'typeof TransferTransaction' is not assignable to parameter of type 'AnyClass` :cry:

Comment (limiaspasdaniel) [reply sdk/src/index.d.ts]: Will check now

Comment (limiaspasdaniel) [reply sdk/src/index.d.ts]: @shuse2 Core is not complaining to me :(. Can you still reproduce this issue?

Comment (shuse2) [reply sdk/src/index.d.ts]: yes, i just tried, but it seems like it is still happening.
```
src/index.ts:47:26 - error TS2345: Argument of type 'typeof TransferTransaction' is not assignable to parameter of type 'AnyClass'.

47  app.registerTransaction(TransferTransaction, {
                            ~~~~~~~~~~~~~~~~~~~

src/index.ts:51:26 - error TS2345: Argument of type 'typeof SecondSignatureTransaction' is not assignable to parameter of type 'AnyClass'.

51  app.registerTransaction(SecondSignatureTransaction, {
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/index.ts:55:26 - error TS2345: Argument of type 'typeof DelegateTransaction' is not assignable to parameter of type 'AnyClass'.

55  app.registerTransaction(DelegateTransaction, {
                            ~~~~~~~~~~~~~~~~~~~

src/index.ts:59:26 - error TS2345: Argument of type 'typeof VoteTransaction' is not assignable to parameter of type 'AnyClass'.

59  app.registerTransaction(VoteTransaction, {
                            ~~~~~~~~~~~~~~~

src/index.ts:63:26 - error TS2345: Argument of type 'typeof MultisignatureTransaction' is not assignable to parameter of type 'AnyClass'.

63  app.registerTransaction(MultisignatureTransaction, {
                            ~~~~~~~~~~~~~~~~~~~~~~~~~

src/index.ts:67:26 - error TS2345: Argument of type 'typeof DappTransaction' is not assignable to parameter of type 'AnyClass'.

67  app.registerTransaction(DappTransaction, {
                            ~~~~~~~~~~~~~~~

src/index.ts:71:26 - error TS2345: Argument of type 'typeof InTransferTransaction' is not assignable to parameter of type 'AnyClass'.

71  app.registerTransaction(InTransferTransaction, {
                            ~~~~~~~~~~~~~~~~~~~~~

src/index.ts:75:26 - error TS2345: Argument of type 'typeof OutTransferTransaction' is not assignable to parameter of type 'AnyClass'.

75  app.registerTransaction(OutTransferTransaction, {
```
18 errors like this

Comment (limiaspasdaniel) [reply sdk/src/index.d.ts]: @shuse2 just fixed it thanks, should work now

**Pull request merged**